### ISSUE TITLE
python cloud client cannot create instances using create

### ISSUE THREAD

**Issue by tazaki** [saltstack/salt#42666]
### Description of Issue/Question

Issues using the salt python api to make instances using salt-cloud 

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

cloud.providers.d/os.conf

openstack:
  minion:
    master: 172.18.222.7
    startup_states: highstate
  identity_url: 'http://192.168.1.225:5000/v3'
  tenant: internal
  user: internal
  password: 'xxxxx'
  driver: nova
  compute_region: RegionOne
  auth_version: 3
  use_keystoneauth: True
  networks:
  - net-id: xxxxxx

cloud.profiles.d/

os_sm_u16s:
  provider: openstack
  size: m1.small
  image: u16-20170726-salt-20170700
  ssh_key_file: /etc/salt/internal.pem
  ssh_key_name: internal
  ssh_interface: private_ips
  ssh_username: ubuntu
  script_args: -CF

NOTE: This profile works fine when using salt-cloud -p os_sm_u16s <name>

and the module:

 salt sm cloud.create openstack test1 image='u16-20170726-salt-20170700' size='m1.small' ssh_username='ubuntu' ssh_key_file='/etc/salt/internal.pem' ssh_key_name='internal' ssh_interface='private_ips' script_args='-CF'

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

Example script:

import salt.cloud
import pprint

cloud_client = salt.cloud.CloudClient(path='/etc/salt/cloud')

myout = cloud_client.create(names=['myinstance1'], provider='openstack',
    kwargs={
        'image': 'u16-20170726-salt-20170700',
        'size': 'm1.small',
        'ssh_username': 'ubuntu',
        'ssh_key_file': '/etc/salt/internal.pem',
        'ssh_key_name': 'internal',
        'ssh_interface': 'private_ips',
        'script_args': '-CF'
    }
)
pprint.pprint(myout)

Example output:

[root@sm python]# python cloud_create2.py
{'myinstance1': {}}

NOTE: It does seem to generate a key?

[root@sm python]# salt-key -L
Accepted Keys:
myinstance1
Denied Keys:
Unaccepted Keys:
Rejected Keys:

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
[root@sm python]# salt --versions-report
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: 1.5.2
       cherrypy: unknown
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: 3.4.3
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.3.1611 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-514.26.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core

**Comment by garethgreenaway** [comment]
The cloud libraries handle communicating with the various cloud providers but in order to manage keys you'll need a connection to the Salt Master, see here for an example: 
https://github.com/saltstack/salt/blob/develop/salt/cloud/cli.py#L39

Is there a reason you're not using the salt-cloud command to accomplish what you're trying to do?

**Comment by tazaki** [comment]
Using salt-cloud with profiles works fine. I just noticed the python api would not work with create and passing the same values.

**Comment by garethgreenaway** [comment]
@tazaki Makes sense and I believe the issue is that you're not establishing a connection to the master to generate the keys.

**Comment by tazaki** [comment]
I am running this on the master. As shown in the example the only thing it seems to do is generate keys:

[root@sm python]# python cloud_create2.py
{'myinstance1': {}}

NOTE: It does seem to generate a key?

[root@sm python]# salt-key -L
Accepted Keys:
myinstance1
Denied Keys:
Unaccepted Keys:
Rejected Keys:

myinstance1 is the machine name im trying to create in the python script using the api. If it was not generating keys would I see it automatically accepted in that list from salt-key -L?
